<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/jlinux/ws_moveit/src/diana_urdf/diana_description/urdf/DianaV2_robot.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="DianaV2">
    <gazebo>
        <plugin filename="libgazebo_ros_control.so" name="ros_control">
            <!--robotNamespace>/</robotNamespace-->
            <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
            <!--legacyModeNS>true</legacyModeNS-->
        </plugin>
        <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
        <plugin name="gazebo_ros_control">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>
    <!-- Kinematic model -->
    <!--
      DH for DianaV2:
      a = [0.0000, 0.0000, 0.0000,  0.0650,  -0.0528,  -0.0122, 0.0870]
      d = [-0.2856,  0.0000,  -0.4586,  0.0000,  -0.4554,  0.0000, -0.1169]
      alpha = [pi, 0.5*pi, -0.5*pi, 0.5*pi, -0.5*pi, -0.5*pi, -0.5*pi]
      q_home_offset = [0, -1.570796327, 0, -1.570796327, 0, 0]
      joint_direction = [1, 1, 1, 1, 1, 1, 1]
      mass = [2.7685, 2.603, 2.2356, 4.735, 1.6367, 2.1397, 2.3561, 0.19295]      
    -->
    <link name="base_link">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.00025974 -0.00026507 0.024973" />
            <mass value="1.61" />
            <inertia ixx="0.0042468" ixy="2.2962E-05" ixz="-7.5575E-06" iyy="0.004245" iyz="-7.4546E-06" izz="0.0072668" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" yz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/base_link.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/base_link.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint1" type="revolute">
        <origin rpy="3.141592653589793 0 0" xyz="0 0 0.2856" />
        <parent link="base_link" />
        <child link="link1" />
        <axis xyz="0 0 1" />
        <limit effort="235.0" lower="-3.12413936106985" upper="3.12413936106985" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link1">
        <inertial>
            <origin rpy="0 0 0" xyz="0.000046 0.001626 0.029490" />
            <mass value="5.094860" />
            <inertia ixx="0.030808" ixy="-0.000013" ixz="-0.000015" iyy="0.028538" iyz="-0.000073" izz="0.014824" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link1.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link1.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint2" type="revolute">
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0" />
        <parent link="link1" />
        <child link="link2" />
        <axis xyz="0 0 1" />
        <limit effort="235.0" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link2">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.000006 -0.048636 -0.035109" />
            <mass value="2.913368" />
            <inertia ixx="0.029103" ixy="-0.000002" ixz="-0.000002" iyy="0.012041" iyz="-0.000023" izz="0.024227" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link2.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link2.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint3" type="revolute">
        <origin rpy="-1.5707963267948966 0 0" xyz="0 -0.4586 0" />
        <parent link="link2" />
        <child link="link3" />
        <axis xyz="0 0 1" />
        <limit effort="111.0" lower="-3.12413936106985" upper="3.12413936106985" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link3">
        <inertial>
            <origin rpy="0 0 0" xyz="0.026887 0.000909 0.141285" />
            <mass value="5.317261" />
            <inertia ixx="0.225449" ixy="-0.000426" ixz="0.001273" iyy="0.233014" iyz="-0.000046" izz="0.023024" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link3.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link3.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint4" type="revolute">
        <origin rpy="1.5707963267948966 0 0" xyz="0.065 0 0" />
        <parent link="link3" />
        <child link="link4" />
        <axis xyz="0 0 1" />
        <limit effort="111.0" lower="0.0" upper="3.0543261909900767" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link4">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.025805 -0.041217 -0.030353" />
            <mass value="1.882670" />
            <inertia ixx="0.013740" ixy="-0.004199" ixz="-0.000032" iyy="0.008787" iyz="0.000021" izz="0.013403" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link4.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link4.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint5" type="revolute">
        <origin rpy="-1.5707963267948966 0 0" xyz="-0.0528 -0.4554 0" />
        <parent link="link4" />
        <child link="link5" />
        <axis xyz="0 0 1" />
        <limit effort="54.0" lower="-3.12413936106985" upper="3.12413936106985" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link5">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.006402 0.023527 0.198019" />
            <mass value="2.671785" />
            <inertia ixx="0.168688" ixy="0.000916" ixz="0.001061" iyy="0.164786" iyz="-0.003773" izz="0.007813" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link5.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link5.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint6" type="revolute">
        <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="-0.0122 0 0" />
        <parent link="link5" />
        <child link="link6" />
        <axis xyz="0 0 1" />
        <limit effort="54.0" lower="-3.12413936106985" upper="3.12413936106985" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link6">
        <inertial>
            <origin rpy="0 0 0" xyz="0.047699 -0.014646 0.000873" />
            <mass value="2.909174" />
            <inertia ixx="0.007828" ixy="0.003751" ixz="-0.000015" iyy="0.017528" iyz="0.000003" izz="0.018824" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link6.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link6.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint7" type="revolute">
        <origin rpy="-1.5707963267948966 0 0" xyz="0.087 -0.1169 0" />
        <parent link="link6" />
        <child link="link7" />
        <axis xyz="0 0 1" />
        <limit effort="54.0" lower="-3.12413936106985" upper="3.12413936106985" velocity="3.141592653589793" />
        <dynamics damping="0.0" friction="0.1" />
    </joint>
    <link name="link7">
        <inertial>
            <origin rpy="0 0 0" xyz="0.000502 0.000000 0.036404" />
            <mass value="0.559484" />
            <inertia ixx="0.001521" ixy="0.000000" ixz="-0.000005" iyy="0.001537" iyz="0.000000" izz="0.000465" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link7.STL" />
            </geometry>
            <material name="LightGrey">
                <color rgba="0.7 0.7 0.7 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/link7.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="diana_hand_joint" type="fixed">
        <parent link="link7" />
        <child link="diana_hand" />
        <origin rpy="0 3.1415926 0" xyz="0 0 0" />
    </joint>
    <link name="diana_hand">
        <visual>
            <geometry>
                <mesh filename="package://diana_description/meshes/hand.dae" />
            </geometry>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://diana_description/meshes/hand.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="diana_leftfinger">
        <visual>
            <geometry>
                <mesh filename="package://diana_description/meshes/finger.dae" />
            </geometry>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://diana_description/meshes/finger.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="diana_rightfinger">
        <visual>
            <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/finger.dae" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://diana_description/meshes/finger.stl" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.1" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <joint name="diana_finger_joint1" type="prismatic">
        <parent link="diana_hand" />
        <child link="diana_leftfinger" />
        <origin rpy="0 0 0" xyz="0 0 0.0584" />
        <axis xyz="0 1 0" />
        <limit effort="20" lower="0.0" upper="0.04" velocity="0.2" />
    </joint>
    <joint name="diana_finger_joint2" type="prismatic">
        <parent link="diana_hand" />
        <child link="diana_rightfinger" />
        <origin rpy="0 0 0" xyz="0 0 0.0584" />
        <axis xyz="0 -1 0" />
        <limit effort="20" lower="0.0" upper="0.04" velocity="0.2" />
        <mimic joint="diana_finger_joint1" />
    </joint>
    <transmission name="joint1_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint1_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <transmission name="joint2_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint2_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <transmission name="joint3_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint3_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <transmission name="joint4_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint4_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <transmission name="joint5_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint5_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <transmission name="joint6_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint6">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint6_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <transmission name="joint7_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint7">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint7_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>
    <gazebo reference="link1">
        <selfCollide>True</selfCollide>
    </gazebo>
    <gazebo reference="link2">
        <selfCollide>True</selfCollide>
    </gazebo>
    <gazebo reference="link3">
        <selfCollide>True</selfCollide>
    </gazebo>
    <gazebo reference="link4">
        <selfCollide>True</selfCollide>
    </gazebo>
    <gazebo reference="link5">
        <selfCollide>True</selfCollide>
    </gazebo>
    <gazebo reference="link6">
        <selfCollide>True</selfCollide>
    </gazebo>
    <gazebo reference="link7">
        <selfCollide>True</selfCollide>
    </gazebo>
    <!-- <xacro:Diana_arm_force prefix="${prefix}" /> -->
    <link name="world" />
    <joint name="world_joint" type="fixed">
        <parent link="world" />
        <child link="base_link" />
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
    </joint>
    <transmission name="trans_diana_finger_joint1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="diana_finger_joint1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="diana_finger_joint1_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_diana_finger_joint2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="diana_finger_joint2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="diana_finger_joint2_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
</robot>

